src/components/resource/edit-managed-title/resource-edit-managed-title.js

```diff
@@ -33,60 +32,36 @@ class ResourceEditManagedTitle extends Component {
     initialValues: PropTypes.object.isRequired,
     intl: intlShape.isRequired,
     model: PropTypes.object.isRequired,
+    onCancel: PropTypes.func.isRequired,
     onSubmit: PropTypes.func.isRequired,
     pristine: PropTypes.bool,
     proxyTypes: PropTypes.object.isRequired
   };
 
-  static contextTypes = {
-    router: PropTypes.shape({
-      history: PropTypes.shape({
-        push: PropTypes.func.isRequired
-      }).isRequired
-    }).isRequired
-  };
-
   state = {
     managedResourceSelected: this.props.initialValues.isSelected,
     showSelectionModal: false,
     allowFormToSubmit: false,
-    formValues: {}
+    formValues: {},
+    initialValues: this.props.initialValues
   }
 
-  componentWillReceiveProps(nextProps) { // eslint-disable-line react/no-deprecated
-    let wasPending = this.props.model.update.isPending && !nextProps.model.update.isPending;
-    let needsUpdate = !isEqual(this.props.model, nextProps.model);
-
-    let wasUnSelected = this.props.model.isSelected && !nextProps.model.isSelected;
-    let isCurrentlySelected = this.props.model.isSelected && nextProps.model.isSelected;
-
-    if (nextProps.initialValues.isSelected !== this.state.managedResourceSelected) {
-      this.setState({
-        managedResourceSelected: nextProps.initialValues.isSelected,
-      });
-    }
-
+  static getDerivedStateFromProps(nextProps, prevState) {
     if (nextProps.model.update.errors.length) {
-      this.setState({
-        showSelectionModal: false
-      });
+      return { showSelectionModal: false };
```

Comment:
I'm concerned that there might be an issue with the `initialValue` logic below never being encountered in this case. Although I suppose if there are errors the `initialValue` stuff doesn't matter?

Reply:
Fixed just in case.